pkg/cmd/server/kubernetes/master.go

```diff
@@ -11,6 +11,7 @@ import (
 
 	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/client/record"
+	osclient "github.com/openshift/origin/pkg/client"
```

Comment:
If the controller were using `cache.Store` (& friends) and/or the controller/informer framework, couldn't each controller be configured during initialization to populate the stores with scale objects based on a field selector to get only kube/origin references?